src/NHibernate/Dialect/DB2Dialect.cs

```diff
@@ -232,28 +232,29 @@ public override bool SupportsVariableLimit
 			get { return false; }
 		}
 
-		public override SqlString GetLimitString(SqlString querySqlString, SqlString offset, SqlString limit)
+		public override SqlString GetLimitString(SqlString sql, SqlString offset, SqlString limit)
 		{
 			if (offset == null)
 			{
-				return new SqlString(querySqlString,
-									 " fetch first ",
-									 limit,
-									 " rows only");
+				return new SqlString(sql, " fetch first ", limit, " rows only");
 			}
 
+			ExtractColumnOrAliasNames(sql, out var selectColumns, out _, out _);
```

Comment:
```suggestion
			var selectColumns = ExtractColumnOrAliasNames(sql);
```

Reply:
This would require lifting `ExtractColumnOrAliasNames` from `OracleXXXDialect` to `DialectBase`

Reply:
Sorry, I had not realized it was not in the base class.